akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -552,6 +569,36 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
         errorCode should ===(ErrorCode.PROTOCOL_ERROR)
       }
     }
+
+    "apply backpressure to request input when responses are not collected" in new TestSetup(Setting(SettingIdentifier.SETTINGS_MAX_CONCURRENT_STREAMS, 5)) with NetProbes {
+      override def settings: ClientConnectionSettings =
+        super.settings.mapHttp2Settings(_.withMaxConcurrentStreams(5))
+
+      def sendOneRequest(streamId: Int): Unit = {
+        user.emitRequest(HttpRequest())
+        network.expectDecodedHEADERS(streamId, endStream = true)
+      }
+
+      user.responseIn.ensureSubscription()
+
+      (1 to 5).foreach(i => sendOneRequest(i * 2 - 1))
+      user.emitRequest(HttpRequest()) // one extra is requested (some buffering?) but not send on the wire
+      network.expectNoBytes(100.millis)
+      user.requestOut.pending shouldBe 0
+      user.requestOut.expectNoMessage(100.millis) // no further request expected
+
+      network.sendHEADERS(3, endStream = false, Seq(RawHeader(":status", "404")))
+      network.expectNoBytes(100.millis)
+      network.sendDATA(3, endStream = true, ByteString.empty)
+      network.expectWindowUpdate()
```

Comment:
Hmmm, where does this WINDOW_UPDATE come from?
I'm guessing it was held back until `3` is closed in the previous line?
Can we add a comment with extra info about that `network.expectWindowUpdate`? Thanks :)